### ISSUE TITLE
Podman rootless exposed port not reachable

### ISSUE THREAD

**Issue by MaserUser** [containers/podman#14174]
/kind bug
**Description**

After recreating the nginx container in a named network space, the exposed port is not reachable any more.
Other containers in the same network can reach the container, so I am sure it is running correctly.
**Steps to reproduce the issue:**

1. podman run -d --replace --name nginx --network web -p 8000:80 nginx:stable

2. curl localhost:8000

**Describe the results you received:**
curl: (7) Failed to connect to localhost port 8000: Connection refused

**Describe the results you expected:**
<!DOCTYPE html>
...
</html>

**Additional information you deem important (e.g. issue happens only occasionally):**
I used the Network ID (web), so that multily pods can communicate with each other, but i do not have to exposed port. Work great before yeaster day.

podman network ls
```
NETWORK ID    NAME        VERSION     PLUGINS
xxxxxxxxxxxx  podman      0.4.0       bridge,portmap,firewall,tuning
xxxxxxxxxxxx  web         0.4.0       bridge,portmap,firewall,tuning,dnsname
```
Tried recreating it,
changed in containers.conf to userns = "auto", so I tried with flag --userns keep-id without any difference.

**Output of `podman version`:**

```
Version:      3.4.2
API Version:  3.4.2
Go Version:   go1.16.6
Built:        Thu Jan  1 01:00:00 1970
OS/Arch:      linux/amd64
```
waiting for ubuntu 22.04.1 to Update, should be latest stable for Ubuntu 20.04

**Output of `podman info --debug`:**

```
host:
  arch: amd64
  buildahVersion: 1.23.1
  cgroupControllers:
  - memory
  - pids
  cgroupManager: systemd
  cgroupVersion: v2
  conmon:
    package: 'conmon: /usr/libexec/podman/conmon'
    path: /usr/libexec/podman/conmon
    version: 'conmon version 2.1.0, commit: '
  cpus: 14
  distribution:
    codename: focal
    distribution: ubuntu
    version: "20.04"
  eventLogger: journald
  hostname: dl380
  idMappings:
    gidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
    uidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
  kernel: 5.4.0-109-generic
  linkmode: dynamic
  logDriver: journald
  memFree: 63543369728
  memTotal: 67172179968
  ociRuntime:
    name: crun
    package: 'crun: /usr/bin/crun'
    path: /usr/bin/crun
    version: |-
      crun version UNKNOWN
      commit: ea1fe3938eefa14eb707f1d22adff4db670645d6
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +CRIU +YAJL
  os: linux
  remoteSocket:
    path: /run/user/1000/podman/podman.sock
  security:
    apparmorEnabled: false
    capabilities: CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
    rootless: true
    seccompEnabled: true
    seccompProfilePath: /usr/share/containers/seccomp.json
    selinuxEnabled: false
  serviceIsRemote: false
  slirp4netns:
    executable: /usr/bin/slirp4netns
    package: 'slirp4netns: /usr/bin/slirp4netns'
    version: |-
      slirp4netns version 1.1.8
      commit: unknown
      libslirp: 4.3.1-git
      SLIRP_CONFIG_VERSION_MAX: 3
      libseccomp: 2.4.3
  swapFree: 8589930496
  swapTotal: 8589930496
  uptime: 2h 21m 46.29s (Approximately 0.08 days)
plugins:
  log:
  - k8s-file
  - none
  - journald
  network:
  - bridge
  - macvlan
  volume:
  - local
registries:
  search:
  - docker.io
  - quay.io
store:
  configFile: /home/MaserUser/.config/containers/storage.conf
  containerStore:
    number: 7
    paused: 0
    running: 7
    stopped: 0
  graphDriverName: overlay
  graphOptions:
    overlay.mount_program:
      Executable: /usr/bin/fuse-overlayfs
      Package: 'fuse-overlayfs: /usr/bin/fuse-overlayfs'
      Version: |-
        fusermount3 version: 3.9.0
        fuse-overlayfs: version 1.5
        FUSE library version 3.9.0
        using FUSE kernel interface version 7.31
  graphRoot: /home/MaserUser/.local/share/containers/storage
  graphStatus:
    Backing Filesystem: extfs
    Native Overlay Diff: "false"
    Supports d_type: "true"
    Using metacopy: "false"
  imageStore:
    number: 8
  runRoot: /run/user/1000/containers
  volumePath: /home/MaserUser/.local/share/containers/storage/volumes
version:
  APIVersion: 3.4.2
  Built: 0
  BuiltTime: Thu Jan  1 01:00:00 1970
  GitCommit: ""
  GoVersion: go1.16.6
  OsArch: linux/amd64
  Version: 3.4.2
```

**Have you tested with the latest version of Podman and have you checked the Podman Troubleshooting Guide? (https://github.com/containers/podman/blob/main/troubleshooting.md)**
Yes.

**Additional environment details (AWS, VirtualBox, physical, etc.):**

Still new to podman.

Thanks for any help
Could not find any solutions to my problem. Guessing that the flag --network bridge solution for Podman v3.2 should be the same as having the new Network ID.

PS: tested it with a new Network ID, still not working, but it still works in the default network.

**Comment by Luap99** [comment]
use --userns=host, rootless port forwarding with a custom userns and networks is broken in that version. This bug is fixed in v4.0.

**Comment by MaserUser** [comment]
Thx alot,
I thought it was fixed in v3.4 and that --userns=keep-id did the same as --userns=host.
but now it is working again.
